From b09c603ca40fc6980188f9c7f419c0918a3f461e Mon Sep 17 00:00:00 2001
From: Gang He <ghe@suse.com>
Date: Wed, 2 May 2018 10:37:48 +0800
Subject: [PATCH 1/3] dlm: fix a clerical error when set SCTP_NODELAY
Git-commit: b09c603ca40fc6980188f9c7f419c0918a3f461e
Patch-mainline: v4.18-rc1
References: bsc#1091594

There is a clerical error when turn off Nagle's algorithm in
sctp_connect_to_sock() function, this results in turn off
Nagle's algorithm failure.
After this correction, DLM performance will be improved obviously
when using SCTP procotol.

Signed-off-by: Gang He <ghe@suse.com>
Signed-off-by: Michal Kubecek <mkubecek@suse.cz>
Signed-off-by: David Teigland <teigland@redhat.com>
---
 fs/dlm/lowcomms.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/dlm/lowcomms.c b/fs/dlm/lowcomms.c
index 5243989a60cc..815125281a86 100644
--- a/fs/dlm/lowcomms.c
+++ b/fs/dlm/lowcomms.c
@@ -1080,7 +1080,7 @@ static void sctp_connect_to_sock(struct connection *con)
 	log_print("connecting to %d", con->nodeid);
 
 	/* Turn off Nagle's algorithm */
-	kernel_setsockopt(sock, SOL_TCP, TCP_NODELAY, (char *)&one,
+	kernel_setsockopt(sock, SOL_SCTP, SCTP_NODELAY, (char *)&one,
 			  sizeof(one));
 
 	result = sock->ops->connect(sock, (struct sockaddr *)&daddr, addr_len,
-- 
2.16.2

